extensions/harvest/src/new.tsx

```diff
@@ -0,0 +1,190 @@
+
+    const project = _.find(projects, (o) => {
+      return o.project.id === parseInt(projectId);
+    });
+    if (typeof project === "object") {
+      setTasks(project.task_assignments);
+      setTaskId(project.task_assignments[0].id.toString());
+    } else {
+      setTasks([]);
+      setTaskId(undefined);
+    }
+  }
+
+  function setTimeFormat(value?: string) {
+    // This function can be called direclty from the onBlur event to better match the Harvest app behavior when it exists
+    if (!value) return;
+
+    if (company?.time_format === "decimal") {
+      if (value.includes(":")) {
+        const parsed = value.split(":");
+        const hour = parseInt(parsed[0]);
+        const minute = parseInt(parsed[1]);
+        if (!isNaN(hour)) {
+          if (!isNaN(minute)) {
+            value = parseFloat(`${hour}.${minute / 60}`)
+              .toFixed(2)
+              .toString();
+          } else {
+            value = hour.toString();
+          }
+        }
+      }
+    }
+    if (company?.time_format === "hours_minutes") {
+      if (!value.includes(":")) {
+        const parsed = parseFloat(value);
+        if (!isNaN(parsed)) {
+          const hour = Math.floor(parsed);
+          const minute = parseInt(((parsed - hour) * 60).toFixed(0));
+          value = `${hour}:${minute < 10 ? "0" : ""}${minute}`;
+        }
+      }
+    }
+    return setHours(value);
+  }
+
+  return (
+    <Form
+      navigationTitle={entry ? "Edit Time Entry" : "New Time Entry"}
+      actions={
+        <ActionPanel>
+          <SubmitFormAction onSubmit={handleSubmit} />
+        </ActionPanel>
+      }
+    >
+      <Form.Dropdown
+        id="project_id"
+        title="Project"
+        value={projectId}
+        onChange={(newValue) => {
+          setProjectId(newValue);
+          setTaskAssignments(newValue);
+        }}
+      >
+        {projects?.map((project) => {
+          return (
+            <Form.DropdownItem
+              value={project.project.id.toString()}
+              title={`[${project.project.code}] ${project.project.name}`}
+              key={project.id}
+            />
+          );
+        })}
+      </Form.Dropdown>
+      <Form.Dropdown id="task_id" title="Task" value={taskId}>
+        {tasks?.map((task) => {
+          return <Form.DropdownItem value={task.task.id.toString()} title={task.task.name} key={task.id} />;
+        })}
+      </Form.Dropdown>
+
+      <Form.Separator />
+
+      <Form.TextArea id="notes" title="Notes" value={notes} onChange={setNotes} />
+      {company?.wants_timestamp_timers && (
+        <>
+          <Form.TextField id="started_time" title="Start Time" placeholder="Leave blank to default to now." />
+          <Form.TextField id="ended_time" title="End Time" placeholder="Leave blank to start a new timer" />
+        </>
+      )}
+      {!company?.wants_timestamp_timers && (
+        <Form.TextField
+          id="hours"
+          title="Duration"
```

Comment:
I'm not sure about naming for **"Duration"**. I tried to provide random letters and it worked. But.. hours / duration sounds like numbers, no? 

What does this field actually for?

Reply:
Duration is the length of the time entry, it's what Harvest calls the field and it actually expects a different format based on how your company has the harvest account setup—either hh:mm or decimal hours.

I'll add some keystroke validation to prevent letters, that should help.